lib/cocoapods-core/specification/dsl.rb

```diff
@@ -76,8 +76,10 @@ module DSL
       #   @param  [String] name
       #           the name of the pod.
       #
-      root_attribute :name,
-                     :required => true
+      attribute :name,
```

Comment:
ah but you shouldn't be able to set the subspecs name directly, it's set via `spec.subspec 'ss_name` do |ss|` bit